### PR TITLE
Added description of checkpointInterval parameter

### PR DESCRIPTION
**Pull request by MrMathias**
Current behavior of ALS and checkpointInterval can result in unexpected behavior, I have added explicit description to hopefully reduce confusion.

## What changes were proposed in this pull request?

better documentation of ml.ALS

## How was this patch tested?

compiled the docs

### CODE REVIEW

**Comment by MrMathias** [comment]
If anyone are interested; checkpointing is only really documented in the scope of Spark streaming, however checkpointing it is extensively used in the ml package.

I think the best solution is to add warnings or even errors to the procedures that uses checkpointing, when a checkpoint directory is not set. Silently ignoring checkpointInterval on heavily iterative procedures that easily causes memory errors, is not a good default behavior.

File: docs/ml-collaborative-filtering.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ by a small set of latent factors that can be used to predict missing entries.
 algorithm to learn these latent factors. The implementation in `spark.ml` has the
 following parameters:
 
+* *checkpointInterval* helps with recovery when nodes fail and StackOverflow exceptions caused by long lineage. **Will be silently ignored if *SparkContext.CheckpointDir* is not set.** (defaults to 10).
```
Comment (felixcheung) [inline docs/ml-collaborative-filtering.md]: the wording is a bit severe... do we have to say node failure or stackoverflow (latter should be rare anyway?)

also is this list of param sorted in any way? perhaps add checkpointInterval to the end?

Comment (MrMathias) [reply docs/ml-collaborative-filtering.md]: Checkpointing exists to better deal with node failure and decrease memory consumption from lineage. This wording is taken from the parameter-comment in the ALS implementation itself, so I think it is fitting.

This list of parameters is both a sub-set and unordered.

Comment (srowen) [reply docs/ml-collaborative-filtering.md]: Nit: StackOverflow exceptions -> either StackOverflowError or stack overflow errors. Also you're nesting `*` and `**` in the markdown; does that work?

**Comment by srowen** [comment]
Ping @MrMathias

**Pull request closed without merging**